rpc/jsonrpc/client/http_json_client.go

```diff
@@ -378,7 +380,13 @@ func makeHTTPDialer(remoteAddr string) (func(string, string) (net.Conn, error),
 	}
 
 	dialFn := func(proto, addr string) (net.Conn, error) {
-		return net.Dial(protocol, u.GetDialAddress())
+		var timeout = 10 * time.Second
+		if !u.isUnixSocket && strings.LastIndex(u.Host, ":") == -1 {
```

Comment:
Do you have an example of input here and resulting output?

Reply:
`https://foo-bar.com` -> `foo-bar.com:https`
`http://foo-bar.com` -> `foo-bar.com:http`
`unix:///tmp/test` -> 'unix:///tmp/test' (because it's UnixSocket)

ref: https://cs.opensource.google/go/go/+/refs/tags/go1.16.4:src/net/ipsock.go;drc=193d5141318d65cea310d995258288bd000d734c;l=248

Reply:
And why do we need to do this?

Reply:
These fixes just try to tackle the issue @marbar3778 posted. We can just close it if we think the current error message is good enough. But the user needs to aware they need to input the URL address with port info.

Reply:
in `newParsedURL` function.if we do `url.Parse` with `https://foo-bar.com`.
We will get `url.Scheme` = "https", `url.Host` = "foo-bar.com"
So under this input case, It's not able to connect to the remote due to a missing port.
Therefore I am padding the protocol at the end of the Host, and then the socket will look up the default port for the protocol.

Reply:
I will let @marbar3778 approve this one 👍

Reply:
I think this makes things a bit easier for when a user is connecting to a URL. What is the norm here? Should the application handle this instead?